pyHS100/smartstrip.py

```diff
@@ -0,0 +1,190 @@
+    print(p.state)
+
+    Errors reported by the device are raised as SmartDeviceExceptions,
+    and should be handled by the user of the library.
+
+    Note:
+    The library references the same structure as defined for the D-Link Switch
+    """
+
+    def __init__(self,
+                 host: str,
+                 protocol: 'TPLinkSmartHomeProtocol' = None) -> None:
+        SmartPlug.__init__(self, host, protocol)
+        self.emeter_type = "emeter"
+        self.plugs = []
+        for plug in self.sys_info["children"]:
+            self.plugs.append(SmartPlug(host, protocol, context=plug["id"]))
+
+    @property
+    def state(self) -> Dict[int, str]:
+        """
+        Retrieve the switch state
+
+        :returns: list with the state of each child plug
+                  SWITCH_STATE_ON
+                  SWITCH_STATE_OFF
+                  SWITCH_STATE_UNKNOWN
+        :rtype: dict
+        """
+        states = {}
+        for plug in range(self.num_children):
+            relay_state = self.sys_info["children"][plug]["state"]
+
+            if relay_state == 0:
+                switch_state = SmartPlug.SWITCH_STATE_OFF
+            if relay_state == 1:
+                switch_state = SmartPlug.SWITCH_STATE_ON
+            else:
+                _LOGGER.warning("Unknown state %s returned.", relay_state)
+                switch_state = SmartPlug.SWITCH_STATE_UNKNOWN
+
+            states[plug] = switch_state
+
+        return states
+
+    @state.setter
+    def state(self, value: str):
+        """
+        Sets the state of all plugs in the strip
+
+        :param value: one of
+                    SWITCH_STATE_ON
+                    SWITCH_STATE_OFF
+        :raises ValueError: on invalid state
+        :raises SmartDeviceException: on error
+
+        """
+        if not isinstance(value, str):
+            raise ValueError("State must be str, not of %s.", type(value))
+        elif value.upper() == SmartPlug.SWITCH_STATE_ON:
+            self.turn_on()
+        elif value.upper() == SmartPlug.SWITCH_STATE_OFF:
+            self.turn_off()
+        else:
+            raise ValueError("State %s is not valid.", value)
+
+    def set_state(self, value: str, *, index: int = -1):
+        """
+        sets the state of a plug on the strip
+
+        :param value: one of
+                    SWITCH_STATE_ON
+                    SWITCH_STATE_OFF
+        :param index: plug index (-1 for all)
+        :raises ValueError: on invalid state
+        :raises SmartDeviceException: on error
+        """
+        self.plugs[index].set_state(value)
+
+    def is_on(self, *, index: int = -1) -> Any:
+        """
+        Returns whether device is on.
+
+        :param index: plug index (-1 for all)
+        :return: True if device is on, False otherwise, Dict without index
+        """
+        if index < 0:
+            is_on = {}
+            for plug in range(self.num_children):
+                is_on[plug] = self.plug[plug].is_on()
```

Comment:
Instead of calling this once per plug, this information should be read once from the sysinfo to avoid unnecessary I/O.